gateways/authorizenet/authorizenet.go

```diff
@@ -134,6 +135,24 @@ func (client *AuthorizeNetClient) RefundWithContext(ctx context.Context, request
 		return nil, err
 	}
 
+	var transactionDetailsResponse *sleet.TransactionDetailsResponse
+	if request.Options != nil && request.Options[sleet.GooglePayTokenOption] != nil {
+		var boolValue bool
+		boolValue, _ = strconv.ParseBool(request.Options[sleet.GooglePayTokenOption].(string))
+		if boolValue {
+			transactionDetailsResponse, err = client.GetTransactionDetails(&sleet.TransactionDetailsRequest{
+				TransactionReference: request.TransactionReference,
+			})
+			if err != nil {
+				return nil, err
+			}
+			//Past last 4 digits of card number
+			creditCardNumber := transactionDetailsResponse.CardNumber
+			last4 := creditCardNumber[len(creditCardNumber)-4:]
+			authorizeNetRefundRequest.CreateTransactionRequest.TransactionRequest.Payment.CreditCard.CardNumber = last4
+		}
```

Comment:
Any nil pointer needs to be checked here?

Reply:
No this is already guaranteed to be built based in the buildRefundRequest function